source/postprocess/global_statistics.cc

```diff
@@ -0,0 +1,296 @@
+
+      if (one_line_per_iteration)
+        for (unsigned int iteration = 0; iteration < nonlinear_iterations; ++iteration)
+          {
+            generate_global_statistics(statistics);
+
+            statistics.add_value("Nonlinear iteration number",
+                                 iteration);
+
+            for (unsigned int column=0; column<advection_solver_controls.size(); ++column)
+              if (iteration < advection_solver_controls[column].second.size())
+                statistics.add_value(advection_solver_controls[column].first,
+                                     advection_solver_controls[column].second[iteration].last_step());
+
+            if (iteration < solver_controls_cheap.size())
+              {
+                statistics.add_value("Iterations for Stokes solver",
+                                     solver_controls_cheap[iteration].last_step() + solver_controls_expensive[iteration].last_step());
+                statistics.add_value("Velocity iterations in Stokes preconditioner",
+                                     list_of_A_iterations[iteration]);
+                statistics.add_value("Schur complement iterations in Stokes preconditioner",
+                                     list_of_S_iterations[iteration]);
+              }
+
+          }
+      else
+        {
+          generate_global_statistics(statistics);
+
+          unsigned int A_iterations = 0;
+          unsigned int S_iterations = 0;
+          unsigned int Stokes_outer_iterations = 0;
+          std::vector<unsigned int> advection_outer_iterations(advection_solver_controls.size(),0);
+
+          for (unsigned int iteration = 0; iteration < nonlinear_iterations; ++iteration)
+            {
+              for (unsigned int column=0; column<advection_solver_controls.size(); ++column)
+                if (iteration < advection_solver_controls[column].second.size())
+                  advection_outer_iterations[column] += advection_solver_controls[column].second[iteration].last_step();
+
+              if (iteration < solver_controls_cheap.size())
+                {
+                  A_iterations += list_of_A_iterations[iteration];
+                  S_iterations += list_of_S_iterations[iteration];
+                  Stokes_outer_iterations += solver_controls_cheap[iteration].last_step() +
+                                             solver_controls_expensive[iteration].last_step();
+                }
+            }
+
+          // only output the number of nonlinear iterations if we actually
+          // use a nonlinear solver scheme
+          const bool output_iteration_number =
+            (!(this->get_parameters().nonlinear_solver == Parameters<dim>::NonlinearSolver::IMPES
+               || this->get_parameters().nonlinear_solver == Parameters<dim>::NonlinearSolver::Advection_only));
+          if (output_iteration_number)
```

Comment:
since you use `output_iteration_number` in only one place, just move the initializer into the `if` condition and be done with it.